docs/contributing.md

```diff
@@ -0,0 +1,115 @@
+# Contributing to the monorepo
+
+## Getting started
+
+- Install [Node.js](https://nodejs.org) version 14.
+  - If you're using [NVM](https://github.com/creationix/nvm#installation) (recommended), `nvm use` will ensure that the right version is installed.
+- Install [Yarn v3](https://yarnpkg.com/getting-started/install).
+- Run `yarn install` to install dependencies and run any required post-install scripts.
+- Run `yarn simple-git-hooks` to add a [Git hook](https://github.com/toplenboren/simple-git-hooks#what-is-a-git-hook) to your local development environment which will ensure that all files pass linting before you push a branch.
+
+## Testing
+
+### Running unit tests
+
+- Run `yarn workspace <workspaceName> run test` to run all tests for a package.
+- Run `yarn workspace <workspaceName> run jest --no-coverage <file>` to run a test file within the context of a package.
+- Run `yarn test` to run tests for all packages.
+
+> **Note**  
+> `workspaceName` here is the `name` field within a package's `package.json`, e.g., `@metamask/announcement-controller`, not the directory where it is located, e.g., `packages/announcement-controller`.
+
+### Integrating with other projects
```

Comment:
Overall these sections and structure are a big step up, great work!

To me this feels more like its own section rather than subtopic of Testing (so `##` rather than `###`).

Nit: Section title could still be more descriptive? Like `Use your fork as a dependency`, `Include custom builds in other projects` or some such other? I guess it's tight to not get too wordy though)

Reply:
> To me this feels more like its own section rather than subtopic of Testing

Sure. It's a distinct process so that makes sense.

> Section title could still be more descriptive

Fair. I was thinking about this in terms of people who may not work with this repo from day to day. For instance, I'm accustomed to working within the extension, but now I've switched to this repo because I've updated some controllers and I've confirmed that the changes work on their own, but I want to go back to the thing that actually uses them and make sure I haven't broken anything. What would you call that step? I was thinking along the lines of manual integration testing. Perhaps it can be thought about differently though. 

In any case, "Include custom builds in other projects" seems closer generally speaking. Maybe "Creating and using on-demand builds in other projects"? I'm not sure I like how technical that would sound, but it _is_ accurate. Hmm...

Reply:
The two last suggestions you listed here seem fine to me! Here are some additional ideas:
 
 * "Using under-development controllers in other projects"
 * "Using controllers in other projects during development/testing"
 * "Testing controllers in other projects"
 * "Testing & local development using controllers in other packages"

Reply:
Okay, I like "Using controllers in other projects during development/testing", I think that's the most accessible version.

Reply:
Renamed in [bae8925](https://github.com/MetaMask/controllers/pull/993/commits/bae89258a3e9936fa0f6b3c4214a4f011cadd749).